File: mysql-async/src/main/scala/com/github/mauricio/async/db/mysql/binary/encoder/ByteBufEncoder.scala
package com.github.mauricio.async.db.mysql.binary.encoder

import com.github.mauricio.async.db.mysql.column.ColumnTypes
import com.github.mauricio.async.db.util.ChannelWrapper.bufferToWrapper
import io.netty.buffer.ByteBuf

object ByteBufEncoder extends BinaryEncoder {
  def encode(value: Any, buffer: ByteBuf) {
    val bytes = value.asInstanceOf[ByteBuf]

    buffer.writeLength(bytes.readableBytes())
    buffer.writeBytes(bytes)
  }

  def encodesTo: Int = ColumnTypes.FIELD_TYPE_BLOB

}

File: mysql-async/src/main/java/com/github/mauricio/async/db/mysql/binary/encoder/ByteBufEncoder.kt
package com.github.mauricio.async.db.mysql.binary.encoder

import com.github.jasync.sql.db.util.writeLength
import com.github.mauricio.async.db.mysql.column.ColumnTypes
import io.netty.buffer.ByteBuf

object ByteBufEncoder : BinaryEncoder {
  override fun encode(value: Any, buffer: ByteBuf) {
    val bytes = value as ByteBuf

    buffer.writeLength(bytes.readableBytes().toLong())
    buffer.writeBytes(bytes)
  }

  override fun encodesTo(): Int = ColumnTypes.FIELD_TYPE_BLOB

}
